crates/ln-dlc-node/src/networking/tungstenite.rs

```diff
@@ -0,0 +1,169 @@
+use crate::networking::DynamicSocketDescriptor;
+use crate::node::NodeInfo;
+use anyhow::Context;
+use futures::future::Either;
+use futures::SinkExt;
+use futures::StreamExt;
+use lightning::ln::peer_handler;
+use lightning::ln::peer_handler::APeerManager;
+use std::future;
+use std::future::Future;
+use std::hash::Hash;
+use std::hash::Hasher;
+use std::ops::ControlFlow;
+use std::ops::Deref;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::UnboundedReceiver;
+use tokio_tungstenite_wasm::Message;
+use tokio_tungstenite_wasm::WebSocketStream;
+use tracing::error;
+
+static ID_COUNTER: AtomicU64 = AtomicU64::new(0);
+
+pub async fn connect_outbound<PM>(
+    peer_manager: PM,
+    node_info: NodeInfo,
+) -> Option<impl Future<Output = ()>>
+where
+    PM: Deref + 'static + Send + Sync + Clone,
+    PM::Target: APeerManager<Descriptor = DynamicSocketDescriptor>,
+{
+    let url = &format!(
+        "ws://{}:{}",
+        node_info.address.ip(),
+        node_info.address.port()
+    );
+    let mut ws = tokio_tungstenite_wasm::connect(url)
+        .await
+        .expect("Can't connect");
```

Comment:
❓Is `expect`ing here dangerous?

Reply:
Ah thanks, this is one of the last two `expect`s that I missed